### PR TITLE
[otel] Initial commit of otel-collector module

### PR DESCRIPTION
**Pull request by jrsdav**
This PR adds OpenTelemetry Collector as an opt-in submodule (or full standalone).

### CODE REVIEW

**Review by EronWright — APPROVED** [review]

File: modules/otel-collector/main.tf [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+          ]
+          exporters = [
+            "googlecloud"
+          ]
+        }
+      }
+    }
+  }
+
+  relay_config = var.cloud_provider == "aws" ? local.aws_relay_conf : (var.cloud_provider == "gcp" ? local.gcp_relay_conf : null)
```
Comment (EronWright) [inline modules/otel-collector/main.tf:139]: I don't understand this "relay" stuff.  I see "exporters" ([ref](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/opentelemetry-operator-0.4.0/charts/opentelemetry-collector/values.yaml#L9)).

Seems to me that the receivers (zipkin, etc) will be the same in all deployments, and the exporters will vary by cloud.  

Hopefully we can drop the `otel_relay_configmap`.

File: modules/otel-collector/main.tf [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    }
+    standaloneCollector = {
+      enabled = true
+      resources = {
+        limits = {
+          cpu    = "500m"
+          memory = "256M"
+        }
+      }
+      # configOverride = {
```
Comment (EronWright) [inline modules/otel-collector/main.tf:177]: I don't understand why you're trying to configure anything using `configOverride`.  I believe that the chart creates the configmap, and we can drop the `otel_relay_configmap`.

Seems to me that the receivers (zipkin, etc) will be the same in all deployments, and the exporters will vary by cloud.

**Review by EronWright — COMMENTED** [review]
This module is probably not useful enough to be in terraform-helm-charts.  Because the exporters are cloud-specific and require an IAM account. 

I have a feeling that `sn_bootstrap` is not a practical module.  You need to perform the composition in `terraform-aws-cloud`, so that you can include a combination of cloud-specific and generic modules.  The `otel-collector` module seems to be of the former variety.

File: modules/otel-collector/main.tf (hunk 2 above)
Comment (jrsdav) [reply modules/otel-collector/main.tf:177]: The advantage of using the `configOverride` value in the helm chart, in any scenario (`sn-bootstrap` module vs composed in `terraform-aws-cloud`), would be the ability for us to do a one-shot apply. If we manage the configmap separately (i.e.`resource "kubernetes_manifest" "otel_relay_configmap" {}`), we would still need to do it _after_ the cluster is created.

Generally, your suggestion around making this cloud provider agnostic makes sense. I'll plan on removing the `cloud_provider` logic before merging this PR, so this module can just be composed in to the respective `terraform-*-cloud` module(s) for any special exporter config. 

In particular, if I can get Helm to feel good when using the the `configOverride` value, making that an input parameter into the module is probably the best way forward.

**Pull request merged**